src/devices/cpu/arcompact/arcompact.h

```diff
@@ -105,795 +40,751 @@ class arcompact_device : public cpu_device
+	bool condition_HI() { return ((!status32_check_c()) && (!status32_check_z())); }
+	// 0x0e - LS - Lower than or same (unsigned)
+	bool condition_LS() { return (status32_check_c() || status32_check_z()); }
+	// 0x0f - PNZ - Positive Non Zero
+	bool condition_PNZ() { return ((!status32_check_n()) && (!status32_check_z())); }
+
+	void check_interrupts();
+
+	/************************************************************************************************************************************
+	*                                                                                                                                   *
+	* 32-bit opcode handlers                                                                                                            *
+	*                                                                                                                                   *
+	************************************************************************************************************************************/
+
+	uint32_t handleop32_B_cc_D_s21(uint32_t op);
+	uint32_t handleop32_B_D_s25(uint32_t op);
+	uint32_t handleop32_BL_cc_d_s21(uint32_t op);
+	uint32_t handleop32_BL_d_s25(uint32_t op);
+
+	int32_t get_01_01_01_address_offset(uint32_t op);
+	uint32_t BRxx_takejump(uint32_t address, uint8_t n, int size);
+	bool BRxx_condition(uint8_t condition, uint32_t b, uint32_t c);
+	uint32_t handleop32_BRxx_reg_reg(uint32_t op, uint8_t condition);
+	uint32_t handleop32_BRxx_reg_imm(uint32_t op, uint8_t condition);
+
+	uint32_t handleop32_BREQ_reg_reg(uint32_t op);
+	uint32_t handleop32_BRNE_reg_reg(uint32_t op);
+	uint32_t handleop32_BRLT_reg_reg(uint32_t op);
+	uint32_t handleop32_BRGE_reg_reg(uint32_t op);
+	uint32_t handleop32_BRLO_reg_reg(uint32_t op);
+	uint32_t handleop32_BRHS_reg_reg(uint32_t op);
+	uint32_t handleop32_BBIT0_reg_reg(uint32_t op);
+	uint32_t handleop32_BBIT1_reg_reg(uint32_t op);
+	uint32_t handleop32_BREQ_reg_imm(uint32_t op);
+	uint32_t handleop32_BRNE_reg_imm(uint32_t op);
+	uint32_t handleop32_BRLT_reg_imm(uint32_t op);
+	uint32_t handleop32_BRGE_reg_imm(uint32_t op);
+	uint32_t handleop32_BRLO_reg_imm(uint32_t op);
+	uint32_t handleop32_BRHS_reg_imm(uint32_t op);
+	uint32_t handleop32_BBIT0_reg_imm(uint32_t op);
+	uint32_t handleop32_BBIT1_reg_imm(uint32_t op);
+	uint32_t handleop32_LD_r_o(uint32_t op);
+	uint32_t handleop32_ST_r_o(uint32_t op);
+
+	typedef uint32_t (*ophandler32)(void*,uint32_t,uint32_t,uint8_t);
+	typedef void (*ophandler32_ff)(void*,uint32_t,uint32_t);
+	typedef void (*ophandler32_mul)(void*,uint32_t,uint32_t);
+	typedef uint32_t (*ophandler32_sop)(void*,uint32_t, uint8_t);
+
+	uint32_t handleop32_general(uint32_t op, ophandler32 ophandler);
+	uint32_t handleop32_general_MULx64(uint32_t op, ophandler32_mul ophandler);
+	uint32_t handleop32_general_nowriteback_forced_flag(uint32_t op, ophandler32_ff ophandler);
+	uint32_t handleop32_general_SOP_group(uint32_t op, ophandler32_sop ophandler);
```

Comment:
Unless you have the body of these in a .ipp file with an `inline` qualifier so it can be available at all points of use as well as the things you pass as handlers, the compiler won’t be able to flatten them for each use case.

Reply:
done